cmd/parca-agent/main.go

```diff
@@ -114,11 +114,19 @@ type Profiler interface {
 }
 
 func main() {
+	hostname, hostnameErr := os.Hostname()
```

Comment:
I wonder if the returned value matches the `nodeName`? https://github.com/parca-dev/parca-agent/blob/5ed98c6e3563fe15078bf5b167bdfd949a574c82/deploy/lib/parca-agent/parca-agent.libsonnet#L303-L312

Reply:
No, it does not, it is the name of the pod. This can be checked in a container with `uname -n`. So passing `nodeName` with the downward API is still needed